protocol/src/protocol.selectionRange.ts

```diff
@@ -73,12 +73,28 @@ export interface SelectionRange {
 	kind: string;
 }
 
+/**
+ * A parameter literal used in selection range requests.
+ */
+export interface SelectionRangeParams {
+	/**
+	 * The text document.
+	 */
+	textDocument: TextDocumentIdentifier;
+
+	/**
+	 * The positions inside the text document.
+	 */
+	positions: Position[];
+}
+
+
 /**
  * A request to provide selection ranges in a document. The request's
  * parameter is of type [TextDocumentPositionParams](#TextDocumentPositionParams), the
- * response is of type [SelectionRange[]](#SelectionRange[]) or a Thenable
+ * response is of type [SelectionRange[][]](#SelectionRange[][]) or a Thenable
  * that resolves to such.
  */
 export namespace SelectionRangeRequest {
-	export const type: RequestType<TextDocumentPositionParams, SelectionRange[] | null, any, any> = new RequestType('textDocument/selectionRange');
+	export const type: RequestType<SelectionRangeParams, SelectionRange[][] | null, any, any> = new RequestType('textDocument/selectionRange');
```

Comment:
I noticed this also. However, many LSP requests allow `null` as a return type so I think it would be weird to not let `null` pass here.

Reply:
Yeah, maybe it’s the Provide*Signature which should be changes

Reply:
You are correct. I removed this by mistake since I saw it with a `ProviderResult` which automatically adds `null`. I will add it back for the request definition

Reply:
Actually commented to early. Everything is fine since `ProviderResult` is defined as:

```
export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>
```

which automatically adds `undefined` and `null`. For request types we need to manually add it.